pilot/pkg/bootstrap/server.go

```diff
@@ -565,14 +564,17 @@ func (s *Server) initIstiodAdminServer(args *PilotArgs, wh *inject.Webhook) erro
 		Addr:    args.ServerOptions.HTTPAddr,
 		Handler: s.httpMux,
 	}
-
+	securedDebug := true
+	if s.httpServer.Addr != "localhost:8080" {
+		securedDebug = false
+	}
 	// create http listener
 	listener, err := net.Listen("tcp", args.ServerOptions.HTTPAddr)
 	if err != nil {
 		return err
 	}
 
-	shouldMultiplex := args.ServerOptions.MonitoringAddr == ""
+	shouldMultiplex := args.ServerOptions.MonitoringAddr == "" && !securedDebug
```

Comment:
I think we have two cases:

Standard: debug on 8080 (localhost), monitoring on 15014, webhook on 15017, ds on 15012
Multiplex: everything on 8080, but disable debug

OR

Multiplex: everything on 8080, but debug has a check like https://github.com/istio/istio/blob/55b2a774e7568ee55d3bc5a0ca0735eda7b5f574/pilot/cmd/pilot-agent/status/server.go#L285

Option (1), disabling debug, seems perfectly fine since now we have the XDS debug

I don't think we want to disable multiplexing; a user is doing multiplexing because they *need* to have everything on one port. So we should be dropping the optional part (debug) not the required part (multiplexing)

Reply:
readiness probe is on 8080 as well. So I am going with this option:  debug has a check for localhost